src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/UpdateContactDetailsServiceTest.java

```diff
@@ -0,0 +1,180 @@
+        caseData.put(INCLUDES_REPRESENTATION_CHANGE, YES_VALUE);
+
+        boolean result = updateContactDetailsService.isIncludesRepresentationChange(caseData);
+
+        assertTrue(result);
+    }
+
+    @Test
+    public void shouldRemoveApplicantSolicitorDetailsIfNocPartyIsApplicant() {
+        Map<String, Object> caseData = new HashMap<>();
+        caseData.put(NOC_PARTY, APPLICANT);
+
+        CaseDetails caseDetails = mock(CaseDetails.class);
+        when(caseDetails.getCaseTypeId()).thenReturn(CaseType.CONSENTED.getCcdType());
+        when(caseDetails.getData()).thenReturn(caseData);
+
+        updateContactDetailsService.handleApplicantRepresentationChange(caseDetails);
+
+
+        assertFalse(caseData.containsKey(APPLICANT_REPRESENTED));
+    }
+
+    @Test
+    public void shouldRemoveRespondentDetailsIfNocPartyIsRespondent() {
+        Map<String, Object> caseData = new HashMap<>();
+        caseData.put(NOC_PARTY, RESPONDENT);
+
+        CaseDetails caseDetails = mock(CaseDetails.class);
+        when(caseDetails.getCaseTypeId()).thenReturn(CaseType.CONSENTED.getCcdType());
+        when(caseDetails.getData()).thenReturn(caseData);
+
+        updateContactDetailsService.handleRespondentRepresentationChange(caseDetails);
+
+        assertFalse(caseData.containsKey(RESPONDENT_ADDRESS));
+    }
+
+    @Test
+    public void shouldRemoveApplicantSolicitorDetailsForContestedCaseWhenNotRepresented() {
+        Map<String, Object> caseData = new HashMap<>();
+        caseData.put(APPLICANT_REPRESENTED, NO_VALUE);
+        caseData.put(CONTESTED_SOLICITOR_NAME, "Solicitor Name");
+        caseData.put(CONTESTED_SOLICITOR_FIRM, "Solicitor Firm");
+        caseData.put(CONTESTED_SOLICITOR_ADDRESS, "Solicitor Address");
+        caseData.put(CONTESTED_SOLICITOR_PHONE, "Solicitor Phone");
+        caseData.put(CONTESTED_SOLICITOR_EMAIL, "Solicitor Email");
+        caseData.put(APP_SOLICITOR_AGREE_TO_RECEIVE_EMAILS_CONTESTED, "Yes");
+        caseData.put(SOLICITOR_REFERENCE, "Reference");
+        caseData.put(APPLICANT_ORGANISATION_POLICY, "Applicant Policy");
+
+        String caseTypeId = CaseType.CONTESTED.getCcdType();
+
+        UpdateContactDetailsService.removeApplicantSolicitorDetails(caseData, caseTypeId);
+
+        assertFalse(caseData.containsKey(CONTESTED_SOLICITOR_NAME));
+        assertFalse(caseData.containsKey(CONTESTED_SOLICITOR_FIRM));
+        assertFalse(caseData.containsKey(CONTESTED_SOLICITOR_ADDRESS));
+        assertFalse(caseData.containsKey(CONTESTED_SOLICITOR_PHONE));
+        assertFalse(caseData.containsKey(CONTESTED_SOLICITOR_EMAIL));
+        assertFalse(caseData.containsKey(APP_SOLICITOR_AGREE_TO_RECEIVE_EMAILS_CONTESTED));
+        assertFalse(caseData.containsKey(SOLICITOR_REFERENCE));
+        assertFalse(caseData.containsKey(APPLICANT_ORGANISATION_POLICY));
+    }
+
+    @Test
+    public void shouldRemoveApplicantSolicitorDetailsForConsentedCaseWhenNotRepresented() {
```

Comment:
Can make use of @ParamterisedTest? It seems the only difference is the `caseTypeId`?

Reply:
[6f9dbad](https://github.com/hmcts/finrem-case-orchestration-service/pull/1860/commits/6f9dbadb1e25a28e3560a5e088e13a78b50e8fc3)